pkg/apis/rhjmc/v1beta1/flightrecorder_types.go

```diff
@@ -58,6 +58,9 @@ type FlightRecorderStatus struct {
 	// Listing of events available in the target JVM
 	// +listType=set
 	Events []EventInfo `json:"events"`
+	// Listing of templates available in the target JVM
+	// +listType=set
```

Comment:
What does +listType=set specify exactly?

Reply:
https://github.com/kubernetes/kube-openapi/blob/master/pkg/idl/doc.go#L21

Reply:
Thanks for bringing this up, actually. I recall us having trouble with a newer Operator SDK due to Events being a Set. I believe this was due to not having a key defined for comparison.

```
The CustomResourceDefinition "flightrecorders.rhjmc.redhat.com" is invalid: spec.validation.openAPIV3Schema.properties[status].properties[events].items.x-kubernetes-map-type: Invalid value: "null": must be atomic as item of a list with x-kubernetes-list-type=set 
```

I think both Events and Templates should be Atomic type instead. They aren't ever updated on a per-element basis, just the entire list updated at once. This would avoid the trouble of not providing a key field. I think it's fine to change both now, while you're making related changes.